core/src/main/java/bisq/core/payment/payload/RevolutAccountPayload.java

```diff
@@ -116,23 +148,30 @@ public String getPaymentDetailsForTradePopup() {
     @Override
     public byte[] getAgeWitnessInputData() {
         // getAgeWitnessInputData is called at new account creation when accountId is empty string.
-        return super.getAgeWitnessInputData(accountId.getBytes(StandardCharsets.UTF_8));
+        if (hasOldAccountId()) {
+            // If the accountId was already in place (updated user who had used accountId for account age) we keep the
+            // old accountId to not invalidate the existing account age witness.
+            return super.getAgeWitnessInputData(accountId.getBytes(StandardCharsets.UTF_8));
+
+        } else {
+            // If a new account was registered from version 1.3.8 or later we use the userName.
+            return super.getAgeWitnessInputData(userName.getBytes(StandardCharsets.UTF_8));
+        }
     }
 
-    public void setUserName(@Nullable String userName) {
-        this.userName = userName;
-        // We only set accountId to userName for new accounts. Existing accounts have accountId set with email
-        // or phone nr. and we keep that to not break account signing.
-        if (accountId.isEmpty()) {
-            accountId = userName;
-        }
+    public boolean userNameNotSet() {
+        return userName.isEmpty();
     }
 
-    public String getUserName() {
-        return userName != null ? userName : accountId;
+    public boolean hasOldAccountId() {
+        return !accountId.equals(userName);
     }
 
-    public boolean userNameNotSet() {
-        return userName == null;
+    public void setUserName(String userName) {
+        this.userName = userName;
+        // We need to set accountId as pre v1.3.8 clients expect the accountId field
+        if (accountId.isEmpty()) {
+            accountId = userName;
+        }
```

Comment:
I just tried to create a new Revolut account and saw that it doesn't behave as expected.

![Bildschirmfoto 2020-09-09 um 16 12 09](https://user-images.githubusercontent.com/170962/92611346-fb2a9e80-f2b8-11ea-95b4-dd9e2dcaef61.png)
![Bildschirmfoto 2020-09-09 um 16 13 35](https://user-images.githubusercontent.com/170962/92611362-fe258f00-f2b8-11ea-8622-50f92e1dfe87.png)

The problem is that it is checked at this place if the accountId is empty which will only work for the first character entered into the field e.g. `B` of `Bobby` this ends up with `accountId` set to `B` and userName set to `Bobby`